libs/application-generic/src/factories/shared/interfaces.ts

```diff
@@ -1,11 +1,91 @@
+import { ChannelTypeEnum, IConfigurations } from '@novu/shared';
 import { IEmailEventBody, ISMSEventBody } from '@novu/stateless';
 
 export interface IHandler {
-  inboundWebhookEnabled?(): boolean;
+  inboundWebhookEnabled(): boolean;
 
-  getMessageId?: (body: any | any[]) => string[];
+  getMessageId: (body: unknown | unknown[]) => string[];
 
-  parseEventBody?: (body: any | any[], identifier: string) => IEmailEventBody | ISMSEventBody | undefined;
+  parseEventBody: (body: unknown | unknown[], identifier: string) => IEmailEventBody | ISMSEventBody | undefined;
 
-  verifySignature?: (body: any, headers: Record<string, string>) => { success: boolean; message?: string };
+  verifySignature: (body: unknown, headers: Record<string, string>) => { success: boolean; message?: string };
+
+  autoConfigureInboundWebhook: (configurations: { webhookUrl: string }) => Promise<{
+    success: boolean;
+    message?: string;
+    configurations?: IConfigurations;
+  }>;
+}
+
+interface IProviderWithWebhookMethods {
+  getMessageId?: (body: unknown | unknown[]) => string[];
+  parseEventBody?: (body: unknown | unknown[], identifier: string) => IEmailEventBody | ISMSEventBody | undefined;
+  verifySignature?: (body: unknown, headers: Record<string, string>) => { success: boolean; message?: string };
+  autoConfigureInboundWebhook?: (configurations: { webhookUrl: string }) => Promise<{
+    success: boolean;
+    message?: string;
+    configurations?: IConfigurations;
+  }>;
+}
+
+export abstract class BaseHandler<T extends IProviderWithWebhookMethods = IProviderWithWebhookMethods>
```

Comment:
now we’ll have a common class responsible for all the agnostic method overrides. its responsibility is to delegate execution to the provider instance, reducing the amount of code in the specific channel-based handlers (i.e BaseSmsHandler, BaseEmailHandler etc).

we need to extract this class into a separate file.